test/document.test.js

```diff
@@ -11928,6 +11928,20 @@ describe('document', function() {
     assert.ok(rawDoc);
     assert.deepStrictEqual(rawDoc.tags, ['mongodb']);
   });
+
+  it('$clone() (gh-11849)', async function() {
+    const schema = new mongoose.Schema({ name: String });
+    const Test = db.model('Test', schema);
+
+    const item = await Test.create({ name: 'Mongoose' });
+
+    const doc = await Test.findById(item._id);
+    const clonedDoc = doc.$clone();
+
+    assert.deepEqual(doc, clonedDoc);
+    assert.deepEqual(doc._doc, clonedDoc._doc);
+    assert.deepEqual(doc.$__, clonedDoc.$__);
```

Comment:
I'd like to add some extra test cases to make sure this `$clone()` method works. For example:

1. Modify a field in `clonedDoc`. Make sure that the field is modified in `clonedDoc`, but **not** in `doc`.
2. Save `clonedDoc`. Make sure that doesn't affect modified paths on `doc`.
3. Add a value that fails validation to `doc`. Run `await doc.validate()`. Then create a new cloned doc `clonedDoc = doc.$clone()`, fix the invalid value, run `await clonedDoc.validate()`. Make sure the path is still invalid in `clonedDoc`.
4. Set a session on `clonedDoc`, make sure that doesn't affect the original doc.